localization/graph_localizer/include/graph_localizer/robust_smart_projection_pose_factor.h

```diff
@@ -75,34 +75,26 @@ class RobustSmartProjectionPoseFactor : public SmartProjectionPoseFactor<CALIBRA
     // Degenerate result tends to lead to solve failures, so return empty factor in this case
     if (result.valid()) {
       this->computeJacobiansSVD(F, E0, b, cameras, *(this->point()));
-      // TODO(rsoussan): This should be here but leads to a degeneracy.  It is done like this in gtsam
-      // for the jacobiansvd factorization but not the hessian factorization -> why?
-    }       // else if (useForRotationOnly(result)) {  // Rotation only factor
-            // Unit3 backProjected = cameras[0].backprojectPointAtInfinity(this->measured().at(0));
-            // this->computeJacobiansSVD(F, E0, b, cameras, backProjected);
-            // }
-    else {  // Empty factor  // NOLINT
+    } else if (useForRotationOnly(result)) {  // Rotation only factor
+      Unit3 backProjected = cameras[0].backprojectPointAtInfinity(this->measured().at(0));
+      this->computeJacobiansSVD(F, E0, b, cameras, backProjected);
+    } else {  // Empty factor  // NOLINT
       return boost::make_shared<JacobianFactorSVD<Dim, 2>>(this->keys());
     }
     return createRegularJacobianFactorSVD<Dim, ZDim>(this->keys(), F, E0, b);
   }
 
   bool useForRotationOnly(const gtsam::TriangulationResult& result) const {
-    if (!rotation_only_fallback_) return false;
-    // Enable some 'invalid' results as these can still be useful for rotation errors
-    // return (result.degenerate());
-    return (result.behindCamera());
+    // Use rotation only for all failure cases
+    return true;
   }
 
   double error(const Values& values) const override {
     if (this->active(values)) {
       try {
         const double total_reprojection_loss = this->totalReprojectionError(this->cameras(values));
         const auto result = this->point();
-        // TODO(rsoussan): This should be here but leads to a degeneracy (see comment in linearize)
-        // if (!result.valid() && !useForRotationOnly(result)) return 0.0;
-        // Multiply by 2 since totalReporjectionError divides mahal distance by 2, and robust_model_->loss
```

Comment:
Did you want to remove the // Multiply by 2 (...) explanation, or deleted by accident?

Reply:
Good catch, I've added this back